src/shogun/multiclass/KNN.cpp

```diff
@@ -100,6 +102,12 @@ bool CKNN::train_machine(CFeatures* data)
 	m_min_label=min_class;
 	m_num_classes=max_class-min_class+1;
 
+	if (m_knn_solver == KNN_KDTREE)
+	{
+		m_kd_tree = new CKDTree(m_leaf_size);
```

Comment:
this is a well tested shogun class?
I remember either kdtree or cover tree was buggy
Also, are both implementation BSD compatible?